opentelemetry/src/trace/span_context.rs

```diff
@@ -212,10 +211,13 @@ impl FromStr for TraceState {
     }
 }
 
+/// A specialized `Result` type for trace state operations.
+type TraceStateResult<T> = Result<T, TraceStateError>;
+
 /// Error returned by `TraceState` operations.
 #[derive(Error, Debug)]
 #[non_exhaustive]
-enum TraceStateError {
+pub enum TraceStateError {
```

Comment:
we need to call out this change in the changelog. All triacetate operations now get a new dedicated Err enum.